Content.Shared/Lock/LockingWhitelistSystem.cs

```diff
@@ -0,0 +1,21 @@
+using Content.Shared.Whitelist;
+
+namespace Content.Shared.Lock;
+
+public sealed class LockingWhitelistSystem : EntitySystem
+{
+    [Dependency] private readonly EntityWhitelistSystem _whitelist = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<LockingWhitelistComponent, LockToggleAttemptEvent>(OnLockToggleAttempt);
+    }
+
+    private void OnLockToggleAttempt(Entity<LockingWhitelistComponent> ent, ref LockToggleAttemptEvent args)
+    {
+        if (args.User == ent.Owner && !_whitelist.IsAllowed(ent, ent.Comp.Blacklist, ent.Comp.Whitelist))
```

Comment:
What is this args.User == ent.Owner check even for? This seems like a weird thing to have in this generic locking comp

Reply:
because the event is raised on both user and the thing have to check which its being raised on, or nobody would be able to unlock borgs